opm/simulators/timestepping/AdaptiveTimeStepping_impl.hpp

```diff
@@ -902,10 +902,14 @@ AdaptiveTimeStepping<TypeTag>::SubStepIteration<Solver>::
 chopTimeStepOrCloseFailingWells_(const int new_time_step)
 {
     bool wells_shut = false;
-    // We are below the threshold, and will check if there are any wells we should close
-    // rather than chopping again.
-    std::set<std::string> failing_wells = detail::consistentlyFailingWells(
-                                         solver_().model().stepReports());
+    // We are below the threshold, and will check if there are any
+    // wells that failes repeatedly (that means that it fails in the last three steps)
+    // we should close rather than chopping again.
+    // If we already have chopped the timestep serveral times due to failed wells we also shut wells that
+    // fails only on this step.
+    bool checkRepeatedFailures = new_time_step > ( minTimeStepBeforeClosingWells_()*restartFactor_()*restartFactor_());
```

Comment:
Why do you multiply twice with the restartFactor_()? It may be ok, but then it should be easy to understand why from the comments in the code.

Reply:
It mimics the checking of the three last timesteps in "repeated failures". 
The idea is that if you already have reduced the timestep 2 times due to well A before it gets shut, you dont want to have to reduce it again for well B. As sometimes well B does not fail until well A is shut. I will add a comment.